<?xml version="1.0"?>

<launch>
  
  <arg name="drone_name"          value="groot"/>
  
  <arg name="bag"                 value="FILE_PATH_HERE"/> 
  
  <!-- play rosbag -->
  <node pkg="rosbag" type="play" name="player" output="screen" args="--clock $(arg bag)"/>

  <!-- Load common parameters -->
  <rosparam command="load" file="$(find roscopter)/params/$(arg drone_name).yaml"/>

  <!-- PID Position Controller -->
  <node name="controller" pkg="roscopter" type="controller">
    <remap from="estimate" to="odom"/>
  </node>

  <!-- Estimator -->
  <!-- Be sure to remap to /PosVelEcef after velocity issue is figured out in ublox -->
  <rosparam command="load" file="$(find roscopter)/params/ekf.yaml" ns="estimator"/>
  <node name="estimator" type="ekf_node" pkg="roscopter" output="screen">
    <remap from="imu" to="/imu/data"/>
    <remap from="baro" to="/baro"/>
    <remap from="ublox_gnss" to="/virtual_PosVelEcef"/>
  </node>

  <!-- Waypoint Manager -->
  <node name="waypoint_manager" pkg="roscopter" type="waypoint_manager.py" output="screen">
    <remap from="waypoint" to="high_level_command"/>
    <remap from="state"    to="odom"/>
  </node>

  <node pkg="rosbag" type="record" name="record" output="screen" args="-O ../data/debug/debug.bag --all"/>
  
</launch>
